src/hardware/serialport/softmodem.cpp

```diff
@@ -102,40 +105,20 @@ static const char *MODEM_GetAddressFromPhone(const char *input) {
 	return nullptr;
 }
 
-CSerialModem::CSerialModem(Bitu id, CommandLine* cmd)
-	: CSerial(id, cmd),
-	  rqueue(new CFifo(MODEM_BUFFER_QUEUE_SIZE)),
-	  tqueue(new CFifo(MODEM_BUFFER_QUEUE_SIZE)),
-	  cmdbuf {0},
-	  commandmode(false),
-	  echo(false),
-	  oldDTRstate(false),
-	  ringing(false),
-	  numericresponse(false),
-	  // Default to direct null modem connection; Telnet mode interprets IAC codes
-	  telnetmode(false),
-	  connected(false),
-	  doresponse(0),
-	  waiting_tx_character(0),
-	  cmdpause(0),
-	  ringtimer(0),
-	  ringcount(0),
-	  plusinc(0),
-	  cmdpos(0),
-	  flowcontrol(0),
-	  tmpbuf {0},
-	  listenport(23),
-	  reg {0},
-	  serversocket(nullptr),
-	  clientsocket(nullptr),
-	  waitingclientsocket(nullptr),
-	  telClient {},
-	  dial {}
+CSerialModem::CSerialModem(const uint8_t port_idx, CommandLine *cmd)
+        : CSerial(port_idx, cmd),
+          rqueue(std::make_unique<CFifo>(MODEM_BUFFER_QUEUE_SIZE)),
+          tqueue(std::make_unique<CFifo>(MODEM_BUFFER_QUEUE_SIZE)),
+          telClient({}),
+          dial({})
 {
 	InstallationSuccessful=false;
 
-	// Setup the listening port, and ignore the return code as this is optional
-	(void) getBituSubstring("listenport:", &listenport, cmd);
+	// Setup the listening port
+	uint32_t val;
+	if (getUintFromString("listenport:", val, cmd))
+		listenport = val;
```

Comment:
Now adds a check if the function succeeded, instead of possibly carring on with a broken or invalid `listenport` value.